homeassistant/components/switch/bbb_gpio.py

```diff
@@ -0,0 +1,92 @@
+"""
+Allows to configure a switch using BBB GPIO.
+
+Switch example for two GPIOs pins P9_12 and P9_42
+Allowed GPIO pin name is GPIOxxx or Px_x
+
+switch:
+  - platform: bbb_gpio
+    ports:
+      GPIO0_7: LED Red
+      P9_12: LED Green
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.switch import PLATFORM_SCHEMA
+import homeassistant.components.bbb_gpio as bbb_gpio
+from homeassistant.const import DEVICE_DEFAULT_NAME
+from homeassistant.helpers.entity import ToggleEntity
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['bbb_gpio']
+
+CONF_PULL_MODE = 'pull_mode'
+CONF_PORTS = 'ports'
+CONF_INVERT_LOGIC = 'invert_logic'
+
+DEFAULT_INVERT_LOGIC = False
+
+_SWITCHES_SCHEMA = vol.Schema({
+    cv.string: cv.string,
```

Comment:
Function from Adafruit_BBIO.GPIO for setup of port takes string as argument. Values of ports are [here](https://camo.githubusercontent.com/a401c1e7d950982582878a90eaea3e7f7afa3f4a/687474703a2f2f7069782e63732e6f6c656d6973732e6564752f637363693538312f4242426c61636b4750494f4d61702e706e67). Mapping of port name/key to specific GPIO is done through table that looks like this

NAME | KEY |  GPIO 
--- | --- | ---
GPIO1_6 | P8_3 | 38
UART4RXD | P9_11 | 30

If from hass perspective positive int is required as port key I assume I could do it through reverse search in mapping table so from int I would find appropriate string. What do you think?